### ISSUE TITLE
[Module Proposal]: `avm-ptn-openai-e2e-baseline`

### ISSUE THREAD

**Issue by cshea-msft** [Azure/Azure-Verified-Modules#1264]
### Check for previous/existing GitHub issues/module proposals

- [x] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [x] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Terraform

### Module Classification?

Pattern Module

### Module Name

avm-ptn-openai-e2e-baseline

### Module Details

I will be mirroring the pattern module in Bicep. https://github.com/Azure/Azure-Verified-Modules/issues/1048

Based off of this doc. https://learn.microsoft.com/en-us/azure/architecture/ai-ml/architecture/baseline-openai-e2e-chat

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

cshea-msft

**Comment by prjelesi** [comment]
@cshea-msft give me a few days to align internally. Tnx!

**Comment by cshea-msft** [comment]
@prjelesi any update on this?

**Comment by prjelesi** [comment]
@cshea-msft can we include 'chat' in module name?

**Comment by cshea-msft** [comment]
@prjelesi no this is what I am following. https://github.com/Azure/Azure-Verified-Modules/issues/1048 i reached out to the owner of the module and said it was different than what we are doing.

**Comment by cshea-msft** [comment]
@prjelesi 

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

**Comment by cshea-msft** [comment]
@prjelesi are we good to move this into in development ?

**Comment by prjelesi** [comment]
@cshea-msft 

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Before starting development, please ensure ALL the following requirements are met.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - for naming your module
  - `TelemetryIdPrefix` - for your module's [telemetry](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr3---category-telemetry---deploymentusage-telemetry)
- For your module's repository:
  - Repo name and folder path are defined in `RepoURL`
  - Create GitHub teams for module owners and contributors and grant them permissions as outlined [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).
  - Grant permissions for the AVM core team and PG teams on your GitHub repo as described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions).

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team

**Comment by mbilalamjad** [comment]
Closing this in favor of https://github.com/Azure/Azure-Verified-Modules/issues/1145 which is already in development
